examplescripts/gitea_deploy.py

```diff
@@ -0,0 +1,382 @@
+from jumpscale.clients.explorer.models import TfgridDeployed_reservation1, NextAction
+from jumpscale.clients.stellar.stellar import Network as StellarNetwork
+
+
+def wait_payment(self, bot, rid, reservation_create_resp=None):
+    """wait slide untill payment is ready
+        Args:
+            bot (GedisChatBot): bot instance
+            rid (int): customer tid
+            threebot_app (bool, optional): is using threebot app payment. Defaults to False.
+            reservation_create_resp (jumpscale.clients.explorer.models.TfgridWorkloadsReservation1, optional): reservation object response. Defaults to None.
+        """
+
+    # wait to check payment is actually done next_action changed from:PAY
+    def is_expired(reservation):
+        return reservation.data_reservation.expiration_provisioning.timestamp() < j.data.time.get().timestamp
+
+    reservation = self._explorer.reservations.get(rid)
+    while True:
+        remaning_time = j.data.time.get(reservation.data_reservation.expiration_provisioning).humanize(
+            granularity=["minute", "second"]
+        )
+        deploying_message = f"""
+Payment being processed...\n
+Deployment will be cancelled if payment is not successful {remaning_time}
+"""
+        print(deploying_message)
+        if reservation.next_action != "PAY":
+            return
+        if is_expired(reservation):
+            res = f"# Failed to wait for payment for reservation:```{reservation.id}```:\n"
+            for x in reservation.results:
+                if x.state == "ERROR":
+                    res += f"\n### {x.category}: ```{x.message}```\n"
+            link = f"{self._explorer.url}/reservations/{reservation.id}"
+            res += f"<h2> <a href={link}>Full reservation info</a></h2>"
+            j.sals.zos.reservation_cancel(rid)
+            break
+        time.sleep(5)
+        reservation = self._explorer.reservations.get(rid)
+
+
+def _check_payment(self, resv_id, currency, total_amount, timeout=300):
+    """Returns True if user has paied alaready, False if not
+        """
+    now = datetime.datetime.now()
+    effects_sum = 0
+    while now + datetime.timedelta(seconds=timeout) > datetime.datetime.now():
+        transactions = self.wallet.list_transactions()
+        for transaction in transactions:
+            if transaction.memo_text == f"{resv_id}":
+                effects = self.wallet.get_transaction_effects(transaction.hash)
+                for e in effects:
+                    if e.asset_code == currency:
+                        effects_sum += e.amount
+        if effects_sum >= total_amount:
+            return True
+    return False
+
+
+def get_ip_range():
+    first_digit = random.choice([172, 10])
+    if first_digit == 10:
+        second_digit = random.randint(0, 255)
+    else:
+        second_digit = random.randint(16, 31)
+    return str(first_digit) + "." + str(second_digit) + ".0.0/16"
+
+
+class Network:
```

Comment:
If we are using jsng why re-implement this one
maybe `from jumpscale.sals.reservation_chatflow.reservation_chatflow import Network`